### PR TITLE
feat(latex): capture placeholder as variable

### PR DESCRIPTION
**Pull request by jdujava**

### CODE REVIEW

**Comment by clason** [comment]
Can you show examples (text and screenshot before/after)?

**Comment by jdujava** [comment]
Before:
image
After:
image

**Comment by clason** [comment]
Hmm, I don't think that's correct. `@variable.parameter` is for the parameters of a function in the signature, not the body (compare C). These should just be `@variable`.

**Comment by jdujava** [comment]
When working with `lua`, I see for example this
image

I thought that it would be nice to have it similarly for `latex`. My colorscheme doesn't do anything special for `@variable` (I think this is usually the default behavior), and I don't want to change this behavior for all languages.

Since `#num` is really special in `latex`, isn't it fine to make an "exeption" and capture it as `@variable.parameter`, or do you propose another solution to obtain nice highlighting of `latex` placeholders?

**Comment by clason** [comment]
The highlight you show is not from tree-sitter, though, but from LSP semantic tokens (which are different and out of our control). In general, we frown upon exceptions; consistent use of captures is the core value that this project provides.

Maybe you can find another capture group that would fit? We can interpret broadly for such different languages as LaTeX, but we won't directly contradict the intended usage (otherwise consistent colorschemes become impossible).

**Comment by jdujava** [comment]
What do you think about `@variable.parameter.builtin`?

**Comment by clason** [comment]
That has the same problem (this is not a parameter).

**Comment by jdujava** [comment]
Sorry, I thought that would mean something different.. Maybe "@punctuation.special"?

**Comment by clason** [comment]
The captures are hierarchical: subcaptures are always a special case. Punctuation doesn't fit, either; it's really just a variable...

My suggestion would be to use `@variable` (which is the semantically correct capture) and just adding `@variable.latex` to your colorscheme.

**Comment by jdujava** [comment]
Nice, I didn't know about such a workaround!

File: queries/latex/highlights.scm [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ; General syntax
 (command_name) @function
 
+(placeholder) @variable.parameter
```
Comment (clason) [inline queries/latex/highlights.scm:4]:
```suggestion
(placeholder) @variable
```

**Pull request merged**